### PR TITLE
[RFR] Fix Forms Remove Empty Values

### LINKED ISSUE #3551: Form ignores empty string values
**What you were expecting:**
Submitting a form with an empty string value submits that value along with the form.

**What happened instead:**
A `TextInput` value == "" is ignored altogether when submitting the form.

**Steps to reproduce:**
Create a `SimpleForm` with a `TextInput` field. Edit the field, and then delete all text (if you leave it empty, it _will_ submit a null value). Submit the form and observe the params that are sent in the request. The empty field is not present at all.

**Related code:**
Will try to get a sandbox posted shortly w/ code.

**Environment**

* React-admin version: v3.0.0-alpha.2
* Last version that did not exhibit the issue (if applicable): 2.9.5
* React version: 16.9.0
* Browser: Chrome Version 76.0.3809.100

### PR DESCRIPTION
**Pull request by djhi**
Supersedes #3656
Fixes #3551, #3660

### CODE REVIEW

File: packages/ra-core/src/form/sanitizeEmptyValues.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    // For every field initialy provided, we check wether it value has been removed
+    // and set it explicitly to an empty string
+    const initialValuesWithEmptyFields = Object.keys(initialValues).reduce(
+        (acc, key) => {
+            if (typeof values[key] === 'object' && values[key] !== null) {
+                acc[key] = sanitizeEmptyValues(initialValues[key], values[key]);
+            } else {
+                acc[key] =
+                    typeof values[key] === 'undefined' ? '' : values[key];
+            }
```
Comment (pemoreau) [inline packages/ra-core/src/form/sanitizeEmptyValues.ts]: As far as I understand you consider 3 cases:
- values[key] is undefined, in that case you initialize the value with ""
- values[key] is a non-null object, and you recursively sanitarize the values
- in all other cases, the value is kept as it is
I think you should also test that initialValues[key]!==null otherwise a recursive call may occurs and throw an error (Object.keys(null) is not defined)
Furthermore, if value[key]===nulll you keep the null value, is that the wanted behavior ?
